enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,486 @@
+3. The aforementioned in-cluster entity will signal CEO to relinquish control of etcd by setting CEO's `managedEtcdKind` to `External`. When this happens, CEO removes the etcd pod from the static pod configs. The resource agents for etcd are running from step 2, and they are configured to wait for etcd pods to be gone so they can restart them using Podman.
+4. The installation proceeds as normal once the pods start.
+If for some reason, the etcd pods cannot be started, then the installation will fail. The installer will need to be able to pull logs from the control-plane nodes to provide context for this failure.
+
+Fencing setup is the last important aspect of the cluster installation. In order for the cluster installation to be successful, fencing should be configured and active before we declare the installation successful. Ideally, the fencing secrets should be made available to the control-plane nodes in the initial pacemaker initialization so that fencing can be configured during step 2. There are a few more critical open questions with this:
+1. Should fencing be made active during the installation or should pacemaker start with it disabled and only enable it after being signaled by the in-cluster entity when the cluster installation is detected as successful?
+2. What mechanism will pacemaker use to get access to the secret linked from the BareMetalHost CRD?
+
+If available via the SaaS offering (not confirmed), ZTP may be evaluated as a future offering. This will need further evaluation to ensure passwords are appropriately handled.
+
+Everything else about cluster creation will be an opaque implementation detail not exposed to the user.
+
+#### Day 2 Procedures
+
+As per a standard 3-node control-plane, OpenShift upgrades and `MachineConfig` changes can not be applied when the cluster is in a degraded state.
+Such operations will only proceed when both peers are online and healthy.
+
+The experience of managing a 2-node control-plane should be largely indistinguishable from that of a 3-node one.
+The primary exception is (re)booting one of the peers while the other is offline and expected to remain so.
+
+As in a 3-node control-plane cluster, starting only one node is not expected to result in a functioning cluster.
+Should the admin wish for the control-plane to start, the admin will need to execute a supplied confirmation command on the active cluster node.
+This command will grant quorum to the RHEL-HA components, authorizing it to fence its peer and start etcd as a cluster-of-one in read/write mode.
+Confirmation can be given at any point and optionally make use of SSH to facilitate initiation by an external script.
+
```

Comment:
What tells the RHEL-HA components that MCD is going to reboot the node during upgrade or while applying a new MachineConfig.

Reply:
@clobrano isn't pacemaker configured to intercept reboot events like this as long as graceful reboots are enabled?

Reply:
yes, pacemaker will not initiate any recovery (e.g. fencing etc.) if the node shuts down, or reboots, gracefully